File: paack-go-sdk/types/order_dto.go
package types

import "fmt"

type OrderDTO struct {
	Status                    string         `json:"status"`
	CodCurrency               string         `json:"cod_currency"`
	CodValue                  float64        `json:"cod_value"`
	Customer                  Customer       `json:"customer"`
	DeliveryType              string         `json:"delivery_type"`
	DeliveryAddress           Address        `json:"delivery_address"`
	DeliveryInstructions      string         `json:"delivery_instructions"`
	ExpectedDeliveryTs        ExpectedTs     `json:"expected_delivery_ts"`
	ExpectedPickUpTs          ExpectedTs     `json:"expected_pick_up_ts"`
	ExternalID                string         `json:"external_id"`
	Parcels                   []Parcel       `json:"parcels"`
	OrderDetails              []ExtraDetails `json:"order_details"`
	PickUpAddress             Address        `json:"pick_up_address"`
	PickUpInstructions        string         `json:"pick_up_instructions"`
	ServiceType               string         `json:"service_type"`
	UndeliverableAddress      Address        `json:"undeliverable_address"`
	UndeliverableInstructions string         `json:"undeliverable_instructions"`
	DeliveryModel             string         `json:"delivery_model"`
	InsuredValue              float64        `json:"insured_value"`
	InsuredCurrency           string         `json:"insured_currency"`
	Labels                    string         `json:"labels"`
}

func MapOrderDTO(order Order) (result OrderDTO, err error, source, code string) {

	err, source, code = order.Validate()
	if err != nil {
		return result, fmt.Errorf("%s", err), "order." + source, code
	}

	return OrderDTO{
		ExternalID:                order.ExternalID,
		Customer:                  order.Customer,
		DeliveryAddress:           order.DeliveryAddress,
		ExpectedDeliveryTs:        order.ExpectedDeliveryTs,
		DeliveryInstructions:      order.DeliveryAddress.Instructions,
		DeliveryType:              order.DeliveryType,
		ServiceType:               order.ServiceType,
		Parcels:                   order.Parcels,
		OrderDetails:              order.OrderDetails,
		PickUpAddress:             order.PickUpAddress,
		ExpectedPickUpTs:          order.ExpectedPickUpTs,
		PickUpInstructions:        order.PickUpAddress.Instructions,
		CodValue:                  order.CashOnDelivery.Amount,
		CodCurrency:               order.CashOnDelivery.Currency,
		UndeliverableAddress:      order.UndeliverableAddress,
		UndeliverableInstructions: order.UndeliverableAddress.Instructions,
		InsuredValue:              order.Insured.Amount,
		InsuredCurrency:           order.Insured.Currency,
	}, nil, "", ""
}

File: paack-java-sdk/src/main/java/eu/paack/sdk/api/dto/OrderDTO.java
package eu.paack.sdk.api.dto;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.*;
import lombok.experimental.SuperBuilder;

import java.util.List;
@Data
@ToString
@SuperBuilder(toBuilder=true)
@NoArgsConstructor
@AllArgsConstructor
public class OrderDTO {

    @JsonProperty("external_id")
    private String externalId;
    @JsonProperty("service_type")
    private String serviceType;
    private CustomerDTO customer;
    @JsonProperty("delivery_address")
    private AddressDTO deliveryAddress;
    @JsonProperty("expected_delivery_ts")
    private TimeSlotDTO expectedDeliveryTs;
    private List<ParcelDTO> parcels;
    @JsonProperty("pick_up_address")
    private AddressDTO pickupAddress;
    @JsonProperty("order_details")
    private List<ExtraDetailDTO> orderDetails;
    @JsonProperty("expected_pickup_ts")
    private TimeSlotDTO expectedPickUpTs;
    @JsonProperty("undeliverable_address")
    private AddressDTO undeliverableAddress;
    @JsonProperty("undeliverable_instructions")
    private String undeliverableInstructions;
    @JsonProperty("delivery_instructions")
    private String deliveryInstructions;
    @JsonProperty("pick_up_instructions")
    private String pickUpInstructions;
    @JsonProperty("deliver_type")
    private String deliverType;
    private String status;
    @JsonProperty("cod_currency")
    private String codCurrency;
    @JsonProperty("cod_value")
    private Double codValue;
    @JsonProperty("delivery_model")
    private String deliveryModel;
    private String labels;
    private List<String> clusters;
    private Double insuredValue;
    private String insuredCurrency;

    public abstract static class OrderDTOBuilder<C extends OrderDTO, B extends OrderDTO.OrderDTOBuilder<C, B>> {
        protected B $fillValuesFromParent(OrderDTO instance) {
            $fillValuesFromInstanceIntoBuilder(instance, this);
            return self();
        }
    }
}
